tests/_utils-tests/model.js

```diff
@@ -0,0 +1,359 @@
+				);
+			} );
+
+			it( 'writes selection collapsed in a text', () => {
+				selection.collapse( root, 3 );
+
+				expect( getData( document, 'main', options ) ).to.equal(
+					'<a></a>fo<selection />o<$text bold=true>bar</$text><b></b>'
+				);
+			} );
+
+			it( 'writes selection collapsed at the text boundary', () => {
+				selection.collapse( elA, 'AFTER' );
+
+				expect( getData( document, 'main', options ) ).to.equal(
+					'<a></a><selection />foo<$text bold=true>bar</$text><b></b>'
+				);
+			} );
+
+			it( 'writes selection collapsed at the text boundary', () => {
+				selection.collapse( elA, 'AFTER' );
+
+				expect( getData( document, 'main', options ) ).to.equal(
+					'<a></a><selection />foo<$text bold=true>bar</$text><b></b>'
+				);
+			} );
+
+			it( 'writes selection collapsed at the end of the root', () => {
+				selection.collapse( root, 'END' );
+
+				// Needed due to https://github.com/ckeditor/ckeditor5-engine/issues/320.
+				selection.clearAttributes();
+
+				expect( getData( document, 'main', options ) ).to.equal(
+					'<a></a>foo<$text bold=true>bar</$text><b></b><selection />'
+				);
+			} );
+
+			it( 'writes selection attributes', () => {
+				selection.collapse( root );
+				selection.setAttributesTo( { italic: true, bold: true } );
+
+				expect( getData( document, 'main', options ) ).to.equal(
+					'<selection bold=true italic=true /><a></a>foo<$text bold=true>bar</$text><b></b>'
+				);
+			} );
+
+			it( 'writes selection collapsed selection in a text with attributes', () => {
+				selection.collapse( root, 5 );
+
+				expect( getData( document, 'main', options ) ).to.equal(
+					'<a></a>foo<$text bold=true>b<selection bold=true />ar</$text><b></b>'
+				);
+			} );
+
+			it( 'writes flat selection containing couple of nodes', () => {
+				selection.addRange(
+					Range.createFromParentsAndOffsets( root, 0, root, 4 )
+				);
+
+				expect( getData( document, 'main', options ) ).to.equal(
+					'<selection><a></a>foo</selection><$text bold=true>bar</$text><b></b>'
+				);
+			} );
+
+			it( 'writes flat selection within text', () => {
+				selection.addRange(
+					Range.createFromParentsAndOffsets( root, 2, root, 3 )
+				);
+
+				expect( getData( document, 'main', options ) ).to.equal(
+					'<a></a>f<selection>o</selection>o<$text bold=true>bar</$text><b></b>'
+				);
+			} );
+
+			it( 'writes multi-level selection', () => {
+				selection.addRange(
+					Range.createFromParentsAndOffsets( elA, 0, elB, 0 )
+				);
+
+				expect( getData( document, 'main', options ) ).to.equal(
+					'<a><selection></a>foo<$text bold=true>bar</$text><b></selection></b>'
+				);
+			} );
+
+			it( 'writes backward selection', () => {
+				selection.addRange(
+					Range.createFromParentsAndOffsets( elA, 0, elB, 0 ),
+					true
+				);
+
+				expect( getData( document, 'main', options ) ).to.equal(
+					'<a><selection backward></a>foo<$text bold=true>bar</$text><b></selection></b>'
```

Comment:
PS. The reason why I don't think that documenting how this function work in detail makes no sense is of course that it will change, but also that you use it for tests. If you more or less understand how it works, then it's just a matter of tuning up TCs that you wrote so they expect the correct format. It's jut that the format should not change.